### PR TITLE
HW Spring Data JPA

### PR DESCRIPTION
**Pull request by katrevichdv**

### CODE REVIEW

File: src/main/java/com/example/bookstore/service/impl/BookServiceImpl.java [hunk 1]
```diff
@@ -20,20 +21,39 @@ public BookServiceImpl(BookRepository bookRepository, BookMapper bookMapper) {
...
-    public BookDto getBookById(Long id) {
-        return bookMapper.toDto(bookRepository.findBookById(id)
+    public BookResponseDto getBookById(Long id) {
+        return bookMapper.toDto(bookRepository.findById(id)
                 .orElseThrow(() -> new EntityNotFoundException("Can't find book by id: " + id)));
     }
+
+    @Override
+    public BookResponseDto update(BookRequestDto bookRequestDto, Long id) {
+        if (bookRepository.existsById(id)) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstore/service/impl/BookServiceImpl.java]: Find book by id
use method from mapper bookMapper.updateBookFromDto(bookRequestDto, book);
save book to DB and return dto

File: src/main/java/com/example/bookstore/service/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -20,20 +21,39 @@ public BookServiceImpl(BookRepository bookRepository, BookMapper bookMapper) {
...
+            Book bookToUpdate = bookMapper.toModel(bookRequestDto);
+            bookToUpdate.setId(id);
+            return bookMapper.toDto(bookRepository.save(bookToUpdate));
+        }
+        throw new EntityNotFoundException("Can't find book to update by id: " + id);
+    }
+
+    @Override
+    public void delete(Long id) {
+        if (bookRepository.existsById(id)) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstore/service/impl/BookServiceImpl.java]:
```suggestion
        bookRepository.deleteById(id);
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Well done. Let's fix some comments

File: src/main/java/com/example/bookstore/dto/BookRequestDto.java [hunk 3]
```diff
@@ -3,7 +3,7 @@
 import lombok.Data;
 
 @Data
-public class CreateBookRequestDto {
+public class BookRequestDto {
     private String title;
     private String author;
     private String isbn;
```
Comment (MentorMateAcademy) [inline src/main/java/com/example/bookstore/dto/BookRequestDto.java]: you can use record for this

**Review by MentorMateAcademy — APPROVED** [review]

**Pull request merged**